### PR TITLE
Add 'Blob.rewrite' API wrapper method. 

### PR DESCRIPTION
**Pull request by tseaver**
~~Uses #2507 as a base.~~

~~NOTE: New system tests cannot be exercised until #2509 is merged.~~

Closes: #1960.

/cc @jonparrott

### CODE REVIEW

File: storage/google/cloud/storage/blob.py [hunk 1]
```diff
@@ -655,6 +655,52 @@ def compose(self, sources, client=None):
...
+            query_params=query_params, data=self._properties, headers=headers,
+            _target_object=self)
+        self._set_properties(api_response['resource'])
+        rewritten = int(api_response['totalBytesRewritten'])
+        size = int(api_response['objectSize'])
+
+        if api_response['done']:
+            return None, rewritten, size
+
+        return api_response['rewriteToken'], rewritten, size
```
Comment (dhermes) [inline storage/google/cloud/storage/blob.py]: @tseaver Are we worried about `KeyError`s in any of these lookups?

Comment (tseaver) [inline storage/google/cloud/storage/blob.py]: Empirically, I'm not seeing the system tests fail due to those keys being missing.

Comment (dhermes) [inline storage/google/cloud/storage/blob.py]: SG

**Comment by dhermes** [comment]
LGTM

**Pull request merged**